### ISSUE TITLE
PENDING action not dispatched

### ISSUE THREAD

**Issue by amaurymartiny** [pburtchaell/redux-promise-middleware#153]
## Version Number
4.2.0

## Code Example

```
const setVariable = (variable) => ({
  type: 'SET_VARIABLE',
  payload: new Promise(resolve => {
    setTimeout(() => {
      resolve(variable);
    }, 3000);
  }),
});
```

## Expected Behavior

See the following actions dispatched:
`SET_VARIABLE`, then `SET_VARIABLE_PENDING`, then after 3s, `SET_VARIABLE_FULLFILLED`

## Actual Behavior
`SET_VARIABLE`, then after 3s, `SET_VARIABLE_FULLFILLED`.

**Comment by tomatau** [comment]
SET_VARIABLE shouldn't be dispatched.

Only SET_VARIABLE_PENDING and SET_VARIABLE_FULFILLED.

How have you added the middleware?

**Comment by amaurymartiny** [comment]
Yes, somewhere in the store definition I have `applyMiddleware(promiseMiddleware())`. It works because SET_VARIABLE_FULFILLED gets dispatched, I didn't create this action type.

**Comment by pburtchaell** [comment]
Can you share your store config?

**Comment by amaurymartiny** [comment]
```
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createLogger } from 'redux-logger';
import { autoRehydrate, persistStore } from 'redux-persist';

import rootReducer from './reducers';
import { AppActions } from './ducks/App';

// Creates the store
const configureStore = () => {
  /* ------------- Redux Configuration ------------- */

  const middleware = [];

  // Add redux-logger in dev mode
  if (process.env !== 'production') {
    middleware.push(createLogger());
  }
  const enhancers = [];

  /* ------------- Thunk Middleware ------------- */

  middleware.push(thunk);

  /* ------------- Promise Middleware ------------- */

  middleware.push(promiseMiddleware());

  /* ------------- Assemble All Middlewares ------------- */

  enhancers.push(applyMiddleware(...middleware));

  /* ------------- AutoRehydrate Enhancer ------------- */

  // Add the autoRehydrate enhancer
  enhancers.push(autoRehydrate());

  const store = createStore(rootReducer, compose(...enhancers));

  // Begin periodically persisting the store
  persistStore(store, {
    whitelist: ['auth'],
  }, () => store.dispatch(AppActions.setAppLoaded(true)));

  return store;
};

export default configureStore;
```

**Comment by tomatau** [comment]
That looks correct to me. Apologies but I haven't used the latest version. Can you please try changing the promise value to be for a key called promise in the payload.

from:

```js
const setVariable = (variable) => ({
  type: 'SET_VARIABLE',
  payload: new Promise(resolve => {
    setTimeout(() => {
      resolve(variable);
    }, 3000);
  }),
});
```

to:

```js
const setVariable = (variable) => ({
  type: 'SET_VARIABLE',
  payload: { promise: new Promise(resolve => {
    setTimeout(() => {
      resolve(variable);
    }, 3000);
  }) },
});
```

If the problem is still there after that. Next try installing `redux-promise-middleware@2` to see if the problem still happens in that version too.

I can't see any path in the code that would lead to _FULFILLED being dispatched without a _PENDING first... are you absolutely certain that the first action type has no suffix??

**Comment by amaurymartiny** [comment]
Mysterious. I tried the two ways, I tried version 2, but always the same result:
![screen shot 2017-05-06 at 9 56 37 pm](https://cloud.githubusercontent.com/assets/1293565/25775834/b1d9d14c-32ae-11e7-8e49-046a7f27e27b.png)

I tried to play a bit with the code, and using `dispatch` instead of `next` does dispatch the PENDING action. The APP_LOADED, APP_LOADED_PENDING and APP_LOADED_FULFILLED are all dispatched. I'm not aware of all the repercussions it can have though, so I'm not 100% sure it's a good idea.
